### DISCUSSION TITLE
Power/Current 0 offset, for HWL8012

### DISCUSSION THREAD

**Discussion by spacelama** [arendst/Tasmota#15503] [Q&A] [answered]
I have some of these: https://templates.blakadder.com/kogan-KASPEMHA.html where I swapped the WB2S for some TYWE2S (also some swapped with esp-12s, and those don't seem to be so problematic, so maybe there's something not quite compatible with the TYWS2S swap)

Using VoltageSet CurrentSet and PowerSet, the values do agree with a metre at various higher power levels.

But not all of them seem to be good for power measurement at the 50watts level and below.  Some of them might nearly double the current from 0.11amps to 0.2amps, which then results in P=VI calculation higher than the power being measured, and thus calculates there must be some reactive power (for a purely resistive light bulb).

I think I could solve my problem for these couple of erroneous devices by subtracting a second current offset.  Is there a setting that already does this?  CurrentCal's talk of fine calibration gave me hope, but it just affects the same value CurrentSet effects.

**Comment by sfromis** [comment, marked as answer]
Did you do a [full calibration](https://tasmota.github.io/docs/Power-Monitoring-Calibration/#calibration-procedure), all 3 steps, with a purely resistive load? Until that, the numbers may not work out very well. (CurrentCal is not what I mean).

**Reply by spacelama** [reply]
Yep, I was trying 25W, 150W and 175W (and 0W, and previously 2kW).  Can't get them bearing a linear relationship.  If calibrated with a load over 100W, the one in front of me right now wants to show 25VA, 0.1A, for 0 current usage.  I've tried both BL0937 CF and HLW8012 CF just to be sure, but I think they only affect the scale of the values anyway.
